server/src/main/java/org/elasticsearch/action/CrossProjectResolverUtils.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+package org.elasticsearch.action;
+
+import org.elasticsearch.ResourceNotFoundException;
+import org.elasticsearch.logging.LogManager;
+import org.elasticsearch.logging.Logger;
+import org.elasticsearch.transport.RemoteClusterAware;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import static org.elasticsearch.transport.RemoteClusterAware.LOCAL_CLUSTER_GROUP_KEY;
+import static org.elasticsearch.transport.RemoteClusterAware.REMOTE_CLUSTER_INDEX_SEPARATOR;
+
+public class CrossProjectResolverUtils {
+    private static final Logger logger = LogManager.getLogger(CrossProjectResolverUtils.class);
+    private static final String WILDCARD = "*";
+    private static final String MATCH_ALL = "_ALL";
+    private static final String EXCLUSION = "-";
+
+    public static void maybeRewriteCrossProjectResolvableRequest(
+        RemoteClusterAware remoteClusterAware,
+        AuthorizedProjectsSupplier.AuthorizedProjects targetProjects,
+        IndicesRequest.CrossProjectReplaceable request
+    ) throws ResourceNotFoundException {
+        if (targetProjects == AuthorizedProjectsSupplier.AuthorizedProjects.NOT_CROSS_PROJECT) {
+            logger.info("Cross-project search is disabled or not applicable, skipping request [{}]...", request);
```

Comment:
Lets change all `logger.info` to `logger.debug` in this file. 
Since this is per request it could result in lots of logs. They could be enabled by users who need them.